### ISSUE TITLE
Sometimes SDL doesn't send OnSystemRequest(LOCK_SCREEN_ICON_URL)

### ISSUE THREAD

**Issue by dboltovskyi** [smartdevicelink/sdl_core#3267]
### Bug Report
Sometimes SDL doesn't send `OnSystemRequest(LOCK_SCREEN_ICON_URL)` even if `url` is defined

##### Occurrence
Sometimes (0.5%)

##### Reproduction Steps
1. Make sure `url` is defined in PT in `endpoints` section
2. Start SDL, HMI, connect Mobile device, register new Application
3. Check SDL sends `OnSystemRequest(LOCK_SCREEN_ICON_URL)`

##### Expected Behavior
Notification is sent

##### Observed Behavior
Notification is not sent due to missing `url` reason:
```
ERROR [29 Feb 2020 01:37:22,598][0x7f4faa361700][Commands] /home/developer/sdl/sdl_core/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc:155 commands::mobile::OnSystemRequestNotification::Run: discarding LOCK_SCREEN_ICON_URL request without URL
```

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (ce12181)
* Testing Against: ATF

##### Test Case, Sample Code, and / or Example App
 - ATF Script: [3267_RAI_OnSystemRequest(LOCK_SCREEN_ICON_URL).lua.tar.gz](https://github.com/smartdevicelink/sdl_core/files/4278829/3267_RAI_OnSystemRequest.LOCK_SCREEN_ICON_URL.lua.tar.gz)
This script needs to be run 50-100 times to catch the issue
 - ATF, SDL log: [2020-03-02_20-39-02.tar.gz](https://github.com/smartdevicelink/sdl_core/files/4278821/2020-03-02_20-39-02.tar.gz)

**Comment by dboltovskyi** [comment]
Issue became reproducible after merge fix for https://github.com/smartdevicelink/sdl_core/issues/2932

**Comment by dboltovskyi** [comment]
@theresalech Please triage this issue.
Updates for `core` and `scripts` are ready for PM review.

**Comment by iCollin** [comment]
Do you really see this issue occur 0.5% of runs? I have now run [3267_1_RAI_OnSystemRequest_LOCK_SCREEN_ICON_URL.lua](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/6eb8406f4762eab4337dc76e3890c9de7e700837/test_scripts/Defects/6_1/3267_1_RAI_OnSystemRequest_LOCK_SCREEN_ICON_URL.lua) 50 times (each run tests 200 times) and I have yet to see a failure.

Can you give me any hint on how to reproduce this more easily? For example will this occur more often is I limit the processing power/memory/IO bandwidth core has available?

**Comment by iCollin** [comment]
Closed via #3271
